src/game/BoardState.java

```diff
@@ -11,44 +21,142 @@
 public class BoardState {
 	// Fields
 	public static final int MAX_BOARD_SIZE = 8;
+	@Deprecated
 	private char[][] boardArray = new char[8][8]; // Represents the current state of the board
-	private int turnNumber;
+	private HashMap<Coordinate, AbstractPiece> pieces;
+
+	public BoardState() {
+		this.pieces = new HashMap<Coordinate, AbstractPiece>();
+
+		this.pieces.put(new Coordinate(0, 0), new Cat(Owner.Player1));
+		this.pieces.put(new Coordinate(1, 0), new Dog(Owner.Player1));
+		this.pieces.put(new Coordinate(2, 0), new Horse(Owner.Player1));
+		this.pieces.put(new Coordinate(3, 0), new Camel(Owner.Player1));
+		this.pieces.put(new Coordinate(4, 0), new Elephant(Owner.Player1));
+		this.pieces.put(new Coordinate(5, 0), new Horse(Owner.Player1));
+		this.pieces.put(new Coordinate(6, 0), new Dog(Owner.Player1));
+		this.pieces.put(new Coordinate(7, 0), new Cat(Owner.Player1));
+
+		this.pieces.put(new Coordinate(0, 1), new Rabbit(Owner.Player1));
+		this.pieces.put(new Coordinate(1, 1), new Rabbit(Owner.Player1));
+		this.pieces.put(new Coordinate(2, 1), new Rabbit(Owner.Player1));
+		this.pieces.put(new Coordinate(3, 1), new Rabbit(Owner.Player1));
+		this.pieces.put(new Coordinate(4, 1), new Rabbit(Owner.Player1));
+		this.pieces.put(new Coordinate(5, 1), new Rabbit(Owner.Player1));
+		this.pieces.put(new Coordinate(6, 1), new Rabbit(Owner.Player1));
+		this.pieces.put(new Coordinate(7, 1), new Rabbit(Owner.Player1));
+
+		this.pieces.put(new Coordinate(0, 7), new Cat(Owner.Player2));
+		this.pieces.put(new Coordinate(1, 7), new Dog(Owner.Player2));
+		this.pieces.put(new Coordinate(2, 7), new Horse(Owner.Player2));
+		this.pieces.put(new Coordinate(3, 7), new Camel(Owner.Player2));
+		this.pieces.put(new Coordinate(4, 7), new Elephant(Owner.Player2));
+		this.pieces.put(new Coordinate(5, 7), new Horse(Owner.Player2));
+		this.pieces.put(new Coordinate(6, 7), new Dog(Owner.Player2));
+		this.pieces.put(new Coordinate(7, 7), new Cat(Owner.Player2));
+
+		this.pieces.put(new Coordinate(0, 6), new Rabbit(Owner.Player2));
+		this.pieces.put(new Coordinate(1, 6), new Rabbit(Owner.Player2));
+		this.pieces.put(new Coordinate(2, 6), new Rabbit(Owner.Player2));
+		this.pieces.put(new Coordinate(3, 6), new Rabbit(Owner.Player2));
+		this.pieces.put(new Coordinate(4, 6), new Rabbit(Owner.Player2));
+		this.pieces.put(new Coordinate(5, 6), new Rabbit(Owner.Player2));
+		this.pieces.put(new Coordinate(6, 6), new Rabbit(Owner.Player2));
+		this.pieces.put(new Coordinate(7, 6), new Rabbit(Owner.Player2));
+	}
+
+	// maybe make this private?
+	public BoardState(HashMap<Coordinate, AbstractPiece> pieces) {
+		this.pieces = pieces;
+	}
 
-	public BoardState(char[][] map, int turnNumber) {
-		for(int i=0; i<8; i++){
-			for(int k=0; k<8; k++){
-				this.boardArray[i][k] = map[i][k];
+	// converter
+	@Deprecated
+	public BoardState(char[][] chars, int turns) {
```

Comment:
this printed out java code to convert the 2d character arrays into the new preferred construction. (making a hashmap of coordinates to pieces, then making a board with that, and then a game with the board)